src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const StateCopy = { ...state };
+  const EmptyArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(StateCopy, action.extraData);
+        EmptyArray.push(Object.assign({}, StateCopy));
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete StateCopy[key];
+        }
+        EmptyArray.push(Object.assign({}, StateCopy));
+        break;
+
+      case 'clear':
+        for (const StateKey in StateCopy) {
+          if (StateCopy.hasOwnProperty(StateKey)) {
```

Comment:
The hasOwnProperty method is not necessary in this context. You can remove it to make your code cleaner.